pkg/querier/querier.go

```diff
@@ -541,7 +541,10 @@ func clampMaxTime(ll log.Logger, maxT int64, refT int64, limitDelta time.Duratio
 	}
 	clampedT := math.Min(maxT, refT+limitDelta.Milliseconds())
 
-	logClampEvent(ll, maxT, clampedT, "max", limitName)
+	if clampedT != maxT {
+		logClampEvent(ll, maxT, clampedT, "max", limitName)
```

Comment:
choosing `!=` instead of a directional `<` or `>` here so we can catch in logs if we clamp incorrectly